dice.py

```diff
@@ -0,0 +1,35 @@
+import random
+
+
+class Dice:
+    """
+    A dice. That you can roll. Can have arbitrary objects sides.
+    """
+    def __init__(self, sides):
```

Comment:
Less a strict coding thing and more a design thing: you could consider taking a number of sides, for the common but simple case of the die being n-sided with integers ` to n.

This would, for example, simplify the creation of the diceList in the parser, letting it be something more like `[dice.Dice(i) for i in range(diceNo)]` (or make it a kwarg rather than an arg, so that it'd be something like `[dice.Dice(n=i) for i in range(diceNo)]`).

There's a couple of ways you could handle that in `__init__`, either keeping the arguments separate and handling what happens if you get one, neither or both, or (and this is what I feel like I would do) accepting both modes as `sides` and, if `sides` is not itself an iterable, trying to run `sides = range(1, 'sides' + 1)` (and then catching the exception in case it wasn't an int)